src/interchange/src/avro/decode.rs

```diff
@@ -442,16 +442,20 @@ impl<'a, 'row> AvroDecode for AvroFlatDecoder<'a, 'row> {
             ValueOrReader::Value(val) => {
```

Comment:
I don't totally understand the circumstances under which we fall into this case and thus fail to include the original bytes in the error message, but regardless this PR is still a strict improvement over the current situation.

Reply:
I suspect this can’t actually happen, but I’d need to do a lot of paging in of the avro code to be sure enough of that to be comfortable using unwrap here rather than just storing an option.

Reply:
Actually, it can definitely happen if, for example, a schema has a `default` value that we use, but which can't be packed into a jsonb because it contains numbers that don't fit in f64. We can probably improve this case but I'd need to think about it more.